### PR TITLE
cpp: Restore C++2A Kconfig

### PR DESCRIPTION
**Pull request by alexanderwachter**
Restore the legacy C++2a Kconfig for compilers that
do not support C++20.

### CODE REVIEW

**Review by stephanosio — COMMENTED** [review]
We forgot to update the CMake files in #36853.

Should update the following as well:
https://github.com/zephyrproject-rtos/zephyr/blob/f77d5c5c53123afe09ff9f0adad13bcf24521cdc/cmake/compiler/compiler_flags_template.cmake#L56
https://github.com/zephyrproject-rtos/zephyr/blob/f77d5c5c53123afe09ff9f0adad13bcf24521cdc/cmake/compiler/arcmwdt/compiler_flags.cmake#L129
https://github.com/zephyrproject-rtos/zephyr/blob/ae85da1c90b6bb824091dfe2960b6cf5b443aab2/cmake/compiler/gcc/compiler_flags.cmake#L126

**Comment by stephanosio** [comment]
Hotfix to be added in this PR, marking DNM for now.

File: cmake/compiler/gcc/compiler_flags.cmake [hunk 1]
```diff
@@ -130,7 +130,8 @@ set_property(TARGET compiler-cpp PROPERTY dialect_cpp14 "-std=c++14" "-Wno-regis
 set_property(TARGET compiler-cpp PROPERTY dialect_cpp17 "-std=c++17" "-Wno-register")
 set_property(TARGET compiler-cpp PROPERTY dialect_cpp2a "-std=c++2a"
   "-Wno-register" "-Wno-volatile")
-
+set_property(TARGET compiler-cpp PROPERTY dialect_cpp20 "-std=c++20"
+  "-Wno-register" "-Wno-volatile")
```
Comment (stephanosio) [inline cmake/compiler/gcc/compiler_flags.cmake:134]: Spurious whitespace change.

File: cmake/compiler/arcmwdt/compiler_flags.cmake [hunk 2]
```diff
@@ -126,10 +126,11 @@ endif()
 set_property(TARGET compiler-cpp PROPERTY dialect_cpp98 "-std=c++98")
 set_property(TARGET compiler-cpp PROPERTY dialect_cpp11 "-std=c++11")
 
-#no support of C++14, C++17, C++2a
+#no support of C++14, C++17, C++2a, C++20
 set_property(TARGET compiler-cpp PROPERTY dialect_cpp14 "")
 set_property(TARGET compiler-cpp PROPERTY dialect_cpp17 "")
 set_property(TARGET compiler-cpp PROPERTY dialect_cpp2a "")
+set_property(TARGET compiler-cpp PROPERTY dialect_cpp20 "")
```
Comment (stephanosio) [inline cmake/compiler/arcmwdt/compiler_flags.cmake:133]: Need to add 2b as well.

File: cmake/compiler/compiler_flags_template.cmake [hunk 3]
```diff
@@ -59,6 +59,7 @@ set_property(TARGET compiler-cpp PROPERTY dialect_cpp11)
 set_property(TARGET compiler-cpp PROPERTY dialect_cpp14)
 set_property(TARGET compiler-cpp PROPERTY dialect_cpp17)
 set_property(TARGET compiler-cpp PROPERTY dialect_cpp2a)
+set_property(TARGET compiler-cpp PROPERTY dialect_cpp20)
```
Comment (stephanosio) [inline cmake/compiler/compiler_flags_template.cmake:62]: Same here.

File: cmake/compiler/gcc/compiler_flags.cmake (hunk 1 above)
Comment (stephanosio) [inline cmake/compiler/gcc/compiler_flags.cmake:134]: Same here. need to add 2b.

**Review by stephanosio — CHANGES_REQUESTED** [review]

File: cmake/compiler/arcmwdt/compiler_flags.cmake [hunk 4]
```diff
@@ -126,10 +126,12 @@ endif()
 set_property(TARGET compiler-cpp PROPERTY dialect_cpp98 "-std=c++98")
 set_property(TARGET compiler-cpp PROPERTY dialect_cpp11 "-std=c++11")
 
-#no support of C++14, C++17, C++2a
+#no support of C++14, C++17, C++2a, C++20
```
Comment (stephanosio) [inline cmake/compiler/arcmwdt/compiler_flags.cmake:129]:
```suggestion
#no support of C++14, C++17, C++2a, C++20, C++2b
```

**Review by stephanosio — COMMENTED** [review]
One more change and it should be good to go.

**Review by stephanosio — APPROVED** [review]
LGTM

p.s. (note to self) we should add a test that exercises these different C++ standards in the future.

**Review by cfriedt — APPROVED** [review]

**Pull request merged**